src/command_add.rs

```diff
@@ -13,7 +13,7 @@ pub fn run_command_add(channel: String, paths: &GlobalPaths) -> Result<()> {
     let required_version = &version_db
         .available_channels
         .get(&channel)
-        .ok_or(anyhow!(
+        .ok_or_else(|| anyhow!(
```

Comment:
By using `ok_or_else` with a closure, we avoid instantiating the error before it is needed. It is similar to the `if ... else` vs `ifelse` tradeoff in Julia.